api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -970,6 +970,11 @@ message Cluster {
   // If `connection_pool_per_downstream_connection` is true, the cluster will use a separate
   // connection pool for every downstream connection
   bool connection_pool_per_downstream_connection = 51;
+
+  // If `connection_pool_idle_timeout` is set, the cluster will remove connection pools
+  // that have not had active connections for the specified timeout duration.
+  google.protobuf.Duration connection_pool_idle_timeout = 53
```

Comment:
Can you update docs at https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/connection_pooling?

Reply:
+1, thanks. If there is missing text there about the previous limit functionality (vs. timeout) that you elude to below, it would be good to discuss that also.

Reply:
Docs have been updated to document the various configurable behaviours.